hs_swat_modelinstance/page_processors.py

```diff
@@ -0,0 +1,125 @@
+                             element_id=content_model.metadata.model_input.id if content_model.metadata.model_input else None)
+
+        ext_md_layout = Layout(
+                                AccordionGroup('Model Output (required)',
+                                    HTML("<div class='form-group' id='modeloutput'> "
+                                        '{% load crispy_forms_tags %} '
+                                        '{% crispy model_output_form %} '
+                                     '</div>'),
+                                ),
+
+                                AccordionGroup('Executed By (required)',
+                                     HTML('<div class="form-group" id="executedby"> '
+                                        '{% load crispy_forms_tags %} '
+                                        '{% crispy executed_by_form %} '
+                                     '</div> '),
+                                ),
+
+                                AccordionGroup('Model Objective (required)',
+                                     HTML('<div class="form-group" id="modelobjective"> '
+                                        '{% load crispy_forms_tags %} '
+                                        '{% crispy model_objective_form %} '
+                                     '</div> '),
+                                ),
+
+                                AccordionGroup('Simulation Type',
+                                     HTML('<div class="form-group" id="simulationtype"> '
+                                        '{% load crispy_forms_tags %} '
+                                        '{% crispy simulation_type_form %} '
+                                     '</div> '),
+                                ),
+
+                                AccordionGroup('Model Methods',
+                                     HTML('<div class="form-group" id="modelmethods"> '
+                                        '{% load crispy_forms_tags %} '
+                                        '{% crispy model_methods_form %} '
+                                     '</div> '),
+                                ),
+
+                                AccordionGroup('SWAT Model Parameters',
+                                     HTML('<div class="form-group" id="swatmodelparameters"> '
+                                        '{% load crispy_forms_tags %} '
+                                        '{% crispy swat_model_parameters_form %} '
+                                     '</div> '),
+                                ),
+
+                                AccordionGroup('Model Input',
+                                     HTML('<div class="form-group" id="modelinput"> '
+                                        '{% load crispy_forms_tags %} '
+                                        '{% crispy model_input_form %} '
+                                     '</div> '),
+                                ),
+                        )
```

Comment:
Another case with lots of blocks of code that are basically identical save for one list of names you could loop over.